### PR TITLE
feat: Firm => add DBR rewards

### PR DESCRIPTION
**Pull request by webmass**
## Description

Add the claimable rewards in FiRM protocol for the INV market.

The INV market has DBR rewards

## Checklist

- [ x] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)

## How to test?

Didn't find a way to test locally
User: 0xAAc0aa431c237C2C0B5f041c8e59B3f1a43aC78F
Claimable DBR amount for the INV market should be > 5289 DBR
DBR APR should be ~ 17%, visible here https://www.inverse.finance/firm/INV

## Question
Is there a way to easily add the rewards infos from another app-id and for a specific address (here escrow address of the depositor)?
The case is the cvxCrv market where the user deposits to an escrow and that escrow stakes into Convex.

### CODE REVIEW

File: src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts [hunk 1]
```diff
@@ -108,4 +113,80 @@ export class EthereumInverseFirmLoanContractPositionFetcher extends ContractPosi
...
+        network: this.network,
+        address: this.invAddress,
+      }),
+    ]);
+
+    if (!dbrtoken || !invToken) return [0];
+
+    const invStakedViaDistributor = xinvExRate.mul(dbrDistributorSupply);
+    const dbrYearlyRewardRate = rewardRate.mul(31536000);
+    const dbrInvExRate = dbrtoken.price / invToken.price;
```
Comment (wpoulin) [inline src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts:187]: I don't know how accurate you want the APR to be, but calling price from oracle will most likely result in a more precise number matching the one on your FIRM's UI 
(I'm guessing the app has oracles for base token prices right ?)

**Review by wpoulin — COMMENTED** [review]
How does one get claimable rewards ?

File: src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts (hunk 1 above)
Comment (webmass) [reply src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts:187]: > I don't know how accurate you want the APR to be, but calling price from oracle will most likely result in a more precise number matching the one on your FIRM's UI (I'm guessing the app has oracles for base token prices right ?)

No we use coingecko prices for the DBR APR on the UI, DBR has no oracle price, it's not a collateral or a borrowable asset.

File: src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts [hunk 2]
```diff
@@ -108,4 +113,80 @@ export class EthereumInverseFirmLoanContractPositionFetcher extends ContractPosi
 
     return [supplied, borrowed];
   }
+
+  async getRewardTokenBalances({
```
[the remark this answers is not in the document]
Comment (webmass) [reply src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts:117]: > This function is not used anywhere

what should be used instead?

[the remark this answers is not in the document]
Comment (wpoulin) [reply src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts:117]: This position is extending `ContractPositionTemplatePositionFetcher`, so you will need to return the balance in `getTokenBalancesPerPosition`

[the remark this answers is not in the document]
Comment (webmass) [reply src/apps/inverse-firm/ethereum/inverse-firm.loan.contract-position-fetcher.ts]: Thanks, updated it

**Review by wpoulin — APPROVED** [review]

**Pull request merged**